x-pack/platform/plugins/shared/entity_manager/public/lib/entity_client.ts

```diff
@@ -124,27 +123,23 @@ export class EntityClient {
     return toKqlExpression(kqlExpression);
   }
 
-  getIdentityFieldsValue(
-    entityInstance: {
-      entity: Pick<EntityInstance['entity'], 'identity_fields'>;
-    } & Required<EntityMetadata>
-  ) {
-    const { identity_fields: identityFields } = entityInstance.entity;
-
-    if (!identityFields) {
+  getIdentityFieldsValue({
+    entity,
+  }: {
+    entity: { [key: string]: any } & { entityIdentityFields: { [key: string]: string[] } };
+  }): Record<string, any> {
+    const { entityIdentityFields: identityFields } = entity;
+    if (!Object.keys(identityFields || {}).length) {
       throw new Error('Identity fields are missing');
     }
 
-    return castArray(identityFields).reduce((acc, field) => {
-      const value = field.split('.').reduce((obj: any, part: string) => {
-        return obj && typeof obj === 'object' ? (obj as Record<string, any>)[part] : undefined;
-      }, entityInstance);
-
-      if (value) {
-        acc[field] = value;
-      }
-
+    return Object.values(identityFields).reduce((acc: Record<string, any>, fields) => {
+      fields.forEach((field) => {
+        if (entity?.[field]) {
+          acc[field] = entity[field];
+        }
+      });
       return acc;
-    }, {} as Record<string, string>);
+    }, {} as Record<string, any>);
```

Comment:
could the identity field value be anything other than `string`, `number`, and possibly `null`?